### DISCUSSION TITLE
Sprint 2 Demo Discussion

### DISCUSSION THREAD

**Discussion by Coffee2Bits** [DevOps-MBSE/AaC#89] [Archive]
Since we're approaching the demo for sprint 2 tomorrow, I figured I'd go ahead and start this discussion. I've listed out the closed issues that I think would be interesting to demo, please post your thoughts, suggestions, and changes.

### Demo-able Topics
- Sphinx API documentation
- Plugin command arguments  
- Protobuf Message Generation
- Markdown system design document generation

**Comment by Coffee2Bits** [comment]
For the gen-protobuf plugin here are the highlights I'd aim to hit:

- Review Basic install/commands/args method
- Review the protobuf_flow model (same as flow, but the data messages are more complex to exercise the protobuf message generation)
   - Data messages all have a `metadata` field added which is generated as a separate proto file and imported by all the messages
   - DataA - has an enum value + has a required/non-optional field (metadata)
   - DataB - has only required fields (proto file won't have any optional fields)
   - DataC - has a repeating field `code`
   - DataD - Nothing special here to show
- Explain some of the behavior and how the data messages exercise it
   -  All protobuf fields are generated with the 'optional' keyword unless the field name is in the `required` attribute
   - Field types with `[]` will be generated with the `repeated` keyword
   - All data and enum definitions that are referenced in interface messages are extracted as individual proto files and imported to their respective proto files
   - Protobuf types are defined with a new field `protobuf_type` which is an enum defined by extending the AaC spec

**Comment by cvchaparro** [comment]
## Demo of Sphinx docs

* Show the pipeline stage that builds the API docs
* Download the API docs artifact
* View in a browser

## Demo of gen-design-doc plugin

* Review the command arguments and how to run the command
  ```bash
  aac gen-design-doc -h
  ```
* Look at `model/flow/flow.yaml` to show an example model from which a system design document can be generated.
* Run the command on the `flow.yaml` file:
  ```bash
  aac gen-design-doc model/flow/flow.yaml .
  ```
* View the generated markdown output
* View the output in a browser (in a Github Gist)

## Demo of gen-protobuf plugin

* Show command arguments and how to run it
  ```bash
  aac gen-protobuf -h
  ```
* Look at `model/protobuf_flow/Data.yaml` to show an example of models with protobuf types specified. (see info above for more context)
* Run the command on the `model/protobuf_flow/flow.yaml` system
  ```bash
  aac gen-protobuf model/protobuf_flow/flow.yaml proto/
  ```
* Show the protobuf files

**Comment by jondavid-black** [comment]
We ran out of time and didn't get to demo.  I think we need to add an hour to our demo day to get all the work we're doing fit in.

**Comment by lizzcondrey** [comment]
Closing since this is marked as resolved and the demo is past.
